src/node-opendds.cpp

```diff
@@ -249,6 +255,35 @@ namespace {
   }
 
   void unsubscribe(const Nan::FunctionCallbackInfo<Value>& fci)
+  {
+    if (fci.Length() < 1 || !fci[0]->IsObject()) {
+      Nan::ThrowTypeError("1 argument required");
+      fci.GetReturnValue().SetUndefined();
+      return;
+    }
+
+    // Get the NodeDRListener
+    const Local<Object> dr_js = fci[0]->ToObject();
+    void* const dr_obj = Nan::GetInternalFieldPointer(dr_js, 0);
+    DDS::DataReader* dr = static_cast<DDS::DataReader*>(dr_obj);
+    NodeDRListener* const ndrl = dynamic_cast<NodeDRListener*>(dr->get_listener());
+
+    // Inform the Listener to skip any remaining takes
+    ndrl->unsubscribing();
+
+    // Schedule a call to unsubscribe_now right after this event
+    Nan::HandleScope scope;
+    Local<Object> globals = fci.GetIsolate()->GetCurrentContext()->Global();
+    globals->Set(V8STR("_cb_obj"), fci.This());
+    globals->Set(V8STR("_cb_arg"), fci[0]);
+    RUN(
```

Comment:
Would AysncWorker (from NAN) be cleaner here?
https://github.com/nodejs/nan/blob/master/doc/asyncworker.md#api_nan_async_worker

In particular, I'm thinking of a worker that has an empty Execute and a non-empty HandleOKCallback.
https://github.com/nodejs/nan/blob/master/examples/async_pi_estimate/async.cc

I couldn't find any other references to how to do this without eval; it should be possible -- I'm probably not familiar enough with the terminology that Node.js/NAN use.